Implement place koma

numama opened the pull request:
クリックすると歩を打つ処理を実装した
その他使いそうな関数を実装した

Commit:
[Update] コメントを適切に修正

Commit:
[Update] 不要なCSSのクラスを削除

- 成り駒の処理をどうするか考える
- 少しhoverを追加

Commit:
[Add] 座標からIDを、IDから座標を取得する関数を作成

Commit:
[Add] クリックするとそこに歩が打たれる処理を実装

- セルのIDと座標を相互に変換できるような関数を実装
- そのセルが先手か後手か判別する関数を実装
- イベントリスナーを用いてクリックしたときに歩を打つ処理を実装

machiyashiki commented on main.js:
```diff
@@ -19,10 +20,19 @@ var createKoma = function(position) {
+      // cellクラスを付与
       $element.classList.add("cell");
+      // セルIDを付与
+      $element.id = "cell" + (colIndex * 9 + rowIndex + 1);
```
getIDbyCoordinates作ったのになぜ使わん？

numama replied:
座標がこの時点ではわからない

machiyashiki commented on main.js:
```diff
@@ -32,8 +42,43 @@ var createKoma = function(position) {
+}
+
+// [関数]先手後手、駒の種類、セルIDを引数にとり、そのセルをその種類のコマに替える -> null
+var placeKoma = function(owner, kind, cellId) {
```
3つ目の引数xとyにしたほうがわかりやすい
デバッグでcellId=47とか言われてもどこか分らんくて多分困る

machiyashiki commented on main.js:
```diff
@@ -32,8 +42,43 @@ var createKoma = function(position) {
 
-createKoma(initPosition);
\ No newline at end of file
+createKoma(initPosition);
```
main文は最後に書く

machiyashiki commented on main.js:
```diff
@@ -11,6 +11,7 @@ var initPosition = [
   [{"owner": "white", "kind": "KY"}, {"owner": "white", "kind": "KE"}, {"owner": "white", "kind": "GI"}, {"owner": "white", "kind": "KI"}, {"owner": "white", "kind": "GY"}, {"owner": "white", "kind": "KI"}, {"owner": "white", "kind": "GI"}, {"owner": "white", "kind": "KE"}, {"owner": "white", "kind": "KY"}]
```
駒の名称略すのやめたほうがいいと思う

numama replied:
わかった。でも書き換えるのちょっとめんどいからまた今度やる。

machiyashiki commented on main.js:
```diff
@@ -32,8 +42,43 @@ var createKoma = function(position) {
+  return [x, y];
+}
+// [関数]座標からIDを取得する -> int
+var getIDbyCoordinates = function(x, y) {
```
ID ->Id

Commit:
[Update] コードレビューに従った

- 微妙な変数や関数名のミス・コメントの誤字を修正
- mainの処理を最後に記述した
- クリックした駒がどっちの駒か、もしくは何も置かれてないかも判断
- placeKoma関数の引数を座標に変更
